diff -NBEbwur --strip-trailing-cr --exclude=.svn --exclude=CVS --exclude=.git ZendFramework-1.10.5/library/Zend/Validate/Hostname.php Zend/Validate/Hostname.php
--- ZendFramework-1.10.5/library/Zend/Validate/Hostname.php	2010-02-15 20:00:17.000000000 -0300
+++ Zend/Validate/Hostname.php	2011-02-02 09:01:20.000000000 -0300
@@ -510,8 +510,10 @@
         if ((count($domainParts) > 1) && (strlen($value) >= 4) && (strlen($value) <= 254)) {
             $status = false;
 
+            /*
             $origenc = iconv_get_encoding('internal_encoding');
             iconv_set_encoding('internal_encoding', 'UTF-8');
+             */
             do {
                 // First check TLD
                 $matches = array();
@@ -581,7 +583,8 @@
                                     $length = $this->_idnLength[strtoupper($this->_tld)];
                                 }
 
-                                if (iconv_strlen($domainPart, 'UTF-8') > $length) {
+                                //if (iconv_strlen($domainPart, 'UTF-8') > $length) {
+                                if (strlen($domainPart) > $length) {
                                     $this->_error(self::INVALID_HOSTNAME);
                                 } else {
                                     $checked = true;
@@ -607,7 +610,7 @@
                 }
             } while (false);
 
-            iconv_set_encoding('internal_encoding', $origenc);
+            //iconv_set_encoding('internal_encoding', $origenc);
             // If the input passes as an Internet domain name, and domain names are allowed, then the hostname
             // passes validation
             if ($status && ($this->_options['allow'] & self::ALLOW_DNS)) {
diff -NBEbwur --strip-trailing-cr --exclude=.svn --exclude=CVS --exclude=.git ZendFramework-1.10.5/library/Zend/Validate/StringLength.php Zend/Validate/StringLength.php
--- ZendFramework-1.10.5/library/Zend/Validate/StringLength.php	2010-01-17 16:03:49.000000000 -0300
+++ Zend/Validate/StringLength.php	2011-02-02 09:01:20.000000000 -0300
@@ -233,7 +233,8 @@
         if ($this->_encoding !== null) {
             $length = iconv_strlen($value, $this->_encoding);
         } else {
-            $length = iconv_strlen($value);
+            // wamp doesn't have iconv...
+            $length = strlen($value);   //iconv_strlen($value);
         }
 
         if ($length < $this->_min) {
